### PR TITLE
[NORMAL] Budget report and graph UI

### PR DESCRIPTION
**Pull request by dorji-dev**
- Budget spending visualization UI
- Budget spending graph UI

### CODE REVIEW

File: src/components/budgets/index.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+              </div>
+            );
+          })}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default Budgets;
```
Comment (dorji-dev) [inline src/components/budgets/index.tsx:88]: Screenshot 2024-04-08 at 20 29 20

File: src/components/spending-report/spending-graph.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        series={[{ data, name: "Spending" }]}
+        type="bar"
+        width="100%"
+        height={400}
+      />
+    </div>
+  );
+};
+
+export default SpendingGraph;
```
Comment (dorji-dev) [inline src/components/spending-report/spending-graph.tsx:87]: Screenshot 2024-04-08 at 20 30 10

File: src/components/spending-report/spending-table.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    state: {
+      sorting,
+    },
+  });
+  return (
+    <DataTable table={table} />
+  );
+};
+
+export default SpendingTable;
```
Comment (dorji-dev) [inline src/components/spending-report/spending-table.tsx:89]: Screenshot 2024-04-08 at 20 30 47

File: src/components/spending-report/spending-graph.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  return (
+    <div className="space-y-[20px] mt-[20px]">
+      <h6 className="font-bold">Spending this month</h6>
+      <div className="flex justify-between">
+        <h4 className="text-[16px] md:text-[20px] font-bold">$12,000</h4>
+        <DateRangePicker />
+      </div>
+      <Chart
+        options={{
+          colors: ["#2563eb33"],
```
Comment (DhanBirSunwar) [inline src/components/spending-report/spending-graph.tsx:37]: use color defined in a variable.

**Pull request merged**